video/out/wayland_common.c

```diff
@@ -1997,23 +1997,25 @@ static const struct wp_fractional_scale_v1_listener fractional_scale_listener =
     preferred_scale,
 };
 
-static void log_color_space(struct mp_log *log, const struct pl_color_space *csp)
-{
-    mp_verbose(log,
-        "transfer: %s, primaries: %s\n"
-        "max_cll=%f, max_fall=%f, min_luma=%f, max_luma=%f\n"
-        "raw prims: red.x=%f,   red.y=%f,\n"
-        "           green.x=%f, green.y=%f,\n"
-        "           blue.x=%f,  blue.y=%f,\n"
-        "           white.x=%f, white.y=%f\n",
-        m_opt_choice_str(pl_csp_trc_names,   csp->transfer),
-        m_opt_choice_str(pl_csp_prim_names, csp->primaries),
-        csp->hdr.max_cll,  csp->hdr.max_fall,
-        csp->hdr.min_luma, csp->hdr.max_luma,
-        csp->hdr.prim.red.x,   csp->hdr.prim.red.y,
-        csp->hdr.prim.green.x, csp->hdr.prim.green.y,
-        csp->hdr.prim.blue.x,  csp->hdr.prim.blue.y,
-        csp->hdr.prim.white.x, csp->hdr.prim.white.y);
+static void log_color_space(struct mp_log *log, const struct pl_color_space *csp,
+                            bool log_hdr_metadata)
+{
+    mp_verbose(log, "transfer: %s, primaries: %s\n",
+        m_opt_choice_str(pl_csp_trc_names, csp->transfer),
+        m_opt_choice_str(pl_csp_prim_names, csp->primaries));
+    if (log_hdr_metadata) {
```

Comment:
What not use `pl_primaries_valid()`?

Reply:
Because us setting primaries is dependent on the compositor supporting set_mastering_display_primaries, not them being valid.

If you prefer I can just not use this helper function for this one

Reply:
`video-target-params` property should follow configured VO output. I think we could clear those unused values to ensure it's consistent.

Reply:
Done that, and also for `repr`